### PR TITLE
Add real-time data availability option

### PR DESCRIPTION
**Pull request by wangch079**
## Closes https://github.com/elastic/enterprise-search-team/issues/5086

This PR documents a workaround if customer is asking for (near) real-time data availability.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: docs/CONNECTOR_PROTOCOL.md [hunk 1]
```diff
@@ -19,6 +19,30 @@ Connectors can:
 
 At this stage, our assumption is that one connector will manage one index, and one index will have only one connector associated with it. This may change in the future.
 
+### Data freshness
+
+Data from remote sources are synced via [sync jobs](DEVELOPING.md#syncing), which means there can be data discrepancy between the remote source and Elasticsearch until the next sync job runs.
+
+There are some data sources with streaming option, which allows external systems to capture changes in real-time. This option has been evaluated, and will not in our roadmap in the near future.
```
Comment (artem-shelkovnikov) [inline docs/CONNECTOR_PROTOCOL.md:26]: It's probably good to have a list of these data sources somewhere and reference it from this document. Do you know if we already have one?

File: docs/CONNECTOR_PROTOCOL.md [hunk 2]
```diff
@@ -19,6 +19,30 @@ Connectors can:
...
+
+If (near) real-time data availability is required, the only option is to [implement incremental sync](DEVELOPING.md#how-an-incremental-sync-works) for the data source, and make it run as frequently a possible.
```
Comment (seanstory) [inline docs/CONNECTOR_PROTOCOL.md:28]: This line and the one above read somewhat negatively. Can we spin this to be neutral or positive? Like:

```
In order to maintain (near) real-time data availability, [implement incremental syncing](DEVELOPING.md#how-an-incremental-sync-works) for the data source, with a frequency that matches your needs.
```

File: docs/CONNECTOR_PROTOCOL.md [hunk 3]
```diff
@@ -19,6 +19,30 @@ Connectors can:
...
+
+Kibana won't allow you to configure a schedule more frequently than every hour, but you can do so via Kibana DevTools:
```
Comment (seanstory) [inline docs/CONNECTOR_PROTOCOL.md:30]:
```suggestion
Kibana won't allow you to configure a schedule more frequently than every hour, but you can do so via the Elasticsearch API (through Kibana DevTools, or CUrl, etc):
```

File: docs/CONNECTOR_PROTOCOL.md (hunk 1 above)
Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md:26]: Streaming API for MongoDB, and change notifications for Sharepoint and OneDrive. I add add these two as example and link to the reference doc.

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md]: Added in https://github.com/elastic/connectors-python/pull/1428/commits/decac08da3d18537d749cdd2fbece5c7a7933973

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md]: Removed OneDrive as it's in the connector catalog yet: https://github.com/elastic/connectors-python/pull/1428/commits/fa9bcdc7997c571f7116ad58a29b5b1ea9fb1169

File: docs/CONNECTOR_PROTOCOL.md [hunk 4]
```diff
@@ -19,6 +19,30 @@ Connectors can:
 
 At this stage, our assumption is that one connector will manage one index, and one index will have only one connector associated with it. This may change in the future.
 
+### Data freshness
+
+Data from remote sources are synced via [sync jobs](DEVELOPING.md#syncing), which means there can be data discrepancy between the remote source and Elasticsearch until the next sync job runs.
+
+There are some data sources with streaming option (E.g. [Change Streams](https://www.mongodb.com/docs/manual/changeStreams) for MongoDB, [Change notifications](https://learn.microsoft.com/en-us/graph/webhooks) for Sharepoint), which allows external systems to capture changes in real-time. This option has been evaluated, and will not in our roadmap in the near future.
```
Comment (seanstory) [inline docs/CONNECTOR_PROTOCOL.md:26]: This feels a-typical, to discuss what we have in our roadmap. @leemthompo @danajuratoni is this something we want to advertise here?

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md:26]: Agree. Any suggestion is appreciated.

Comment (leemthompo) [reply docs/CONNECTOR_PROTOCOL.md:26]: 🤔 My instinct would be to remove that.

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md:26]: Hi @leemthompo , did you mean to remove the whole paragraph?

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md]: Updated: https://github.com/elastic/connectors-python/pull/1428/commits/24a8bf63047af4dfd5d26eadcf7114afd1aec2ca

**Review by seanstory — APPROVED** [review]

**Pull request merged**